### ISSUE TITLE
unspecified-encoding should not be raised when PYTHONUTF8=1

### ISSUE THREAD

**Issue by sam-s** [pylint-dev/pylint#8979]
### Current problem

When `PYTHONUTF8=1` is set, the encoding is specified to be UTF-8.
I don't think `unspecified-encoding` is appropriate in such a situation

### Desired solution

no `unspecified-encoding` reported when `PYTHONUTF8=1` or [Python version > 3.15](https://peps.python.org/pep-0686/)

**Comment by DudeNr33** [comment]
I agree that once PEP686 is implemented the message can be silenced for those Python versions. Thanks for the heads-up, this makes sense.

But I would not silence the message just because the environment variable `PYTHONUTF8` is set.

The point of the warning is to explicitly hint the user at potential problems in portability:

> It is better to specify an encoding when opening documents. Using the system default implicitly can create problems **on other operating systems**. See https://peps.python.org/pep-0597/

Just as (in general) you can not rely on the fact that other users are using an environment with the same default encoding as you are, you can not rely on the fact that they will have this environment variable set.

If in your specific case you have direct control over all possible execution environments and know that either the default encoding or environment variables are set as you need it, it would make more sense to disable `unspecified-encoding` in your `.pylintrc`/`pyproject.toml`.
https://github.com/pylint-dev/pylint/blob/42baaea0cb97bf418f06b08438b7864378a7413a/examples/pyproject.toml#L368

**Comment by PanderMusubi** [comment]
It should be `≥ 3.15`, see for that https://github.com/pylint-dev/pylint/issues/10791

**Comment by Pierre-Sassoulas** [comment]
Closing as won't do for the reason given by @DudeNr33 .
